modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/LeaseUpdater.java

```diff
@@ -173,14 +175,19 @@ public void deactivate() {
      * Finds a node that can be the leaseholder.
      *
      * @param assignments Replication group assignment.
+     * @param proposed Proposed consistent id, found out of a lease negotiation. The parameter might be {@code null}.
      * @return Cluster node, or {@code null} if no node in assignments can be the leaseholder.
      */
-    private ClusterNode nextLeaseHolder(Set<Assignment> assignments) {
+    private ClusterNode nextLeaseHolder(Set<Assignment> assignments, @Nullable String proposed) {
```

Comment:
`proposed` sounds like a boolean flag. Could we rename it to anything like `proposedConsistentId`?